src/test/java/org/reactome/addlinks/test/TestEnsemblBioMartFileProcessor.java

```diff
@@ -0,0 +1,69 @@
+package org.reactome.addlinks.test;
+
+import org.junit.Test;
+import org.reactome.addlinks.fileprocessors.ensembl.EnsemblBioMartFileProcessor;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+public class TestEnsemblBioMartFileProcessor {
+
+    private static final String uniprotToProteinsKey = "hsapiens_uniprotToProteins";
+    private static final String transcriptToMicroarraysKey = "hsapiens_transcriptToMicroarrays";
+    private static final String proteinToGenesKey = "hsapiens_proteinToGenes";
+    private static final String proteinToTranscriptsKey = "hsapiens_proteinToTranscripts";
+    private static final int EXPECTED_RESULTS_LENGTH = 50;
+    private static final String testDirectory = "/tmp/test_ensembl_biomart_mapping_service/";
+    private static final Path improperlyFormattedFilepath = Paths.get(testDirectory + "hsapiens_improperly_formatted_microarray");
+    private static final List<String> improperlyFormattedLines = new ArrayList<>(Arrays.asList("ENSG00000174953ENST00000308361ENSP00000309296ILMN_1738272\n", "ENSG00000230791ENST00000414519ENSP00000400138ILMN_1717127\n"));
+
+
+    @Test
+    public void testEnsemblBioMartFileProcessor() throws IOException {
+        EnsemblBioMartFileProcessor processor = new EnsemblBioMartFileProcessor();
+        processor.setPath(Paths.get("src/test/resources/"));
+        Map<String, Map<String, List<String>>> testMapping = processor.getIdMappingsFromFile();
+
+        boolean uniprotToProteinsPopulated = testMapping.containsKey(uniprotToProteinsKey) && testMapping.get(uniprotToProteinsKey).size() == EXPECTED_RESULTS_LENGTH;
+        assertThat(uniprotToProteinsPopulated, is(equalTo(true)));
+        boolean transcriptToMicroarraysPopulated = testMapping.containsKey(transcriptToMicroarraysKey) && testMapping.get(transcriptToMicroarraysKey).size() == EXPECTED_RESULTS_LENGTH;
+        assertThat(transcriptToMicroarraysPopulated, is(equalTo(true)));
+        boolean proteinToGenesPopulated = testMapping.containsKey(proteinToGenesKey) && testMapping.get(proteinToGenesKey).size() == EXPECTED_RESULTS_LENGTH;
+        assertThat(proteinToGenesPopulated, is(equalTo(true)));
+        boolean proteinToTranscriptsPopulated = testMapping.containsKey(proteinToTranscriptsKey) && testMapping.get(proteinToTranscriptsKey).size() == EXPECTED_RESULTS_LENGTH;
+        assertThat(proteinToTranscriptsPopulated, is(equalTo(true)));
```

Comment:
These tests could use specific Hamcrest matchers:

```
assertThat(testMapping, hasKey(uniprotToProteinsKey));
assertThat(testMapping, is(aMapWithSize(EXPECTED_RESULTS_LENGTH)));
```

Optionally, each mapping type tested here could be split into its own smaller test using a common @before annotated method.  Keeping tests as small scoped as possible with as few asserts as possible (ideally only one) is best practice.

```
private EnsemblBioMartFileProcessor ensemblBioMartFileProcessor;

@Before
public void createEnsemblBioMartFileProcessor() {
    ensemblBioMartFileProcessor = new EnsemblBioMartFileProcessor();
}

@Test
public void testEnsemblBioMartFileProcessor() throws IOException {
    ensemblBioMartFileProcessor.setPath(Paths.get("src/test/resources/"));
    Map<String, Map<String, List<String>>> testMapping = ensemblBioMartFileProcessor.getIdMappingsFromFile();

    assertThat(testMapping, hasKey(uniprotToProteinsKey));
    assertThat(testMapping, is(aMapWithSize(EXPECTED_RESULTS_LENGTH)));
}
```

Reply:
Done, and they were split into seperate tests

Reply:
Great, thanks!  For the other assert statements that use `is(equalTo(true))`, would it be helpful to mention the hamcrest matchers that can be used for them in comments below?